src/mapper.py

```diff
@@ -79,9 +80,20 @@ def generate_mapping(self, rtf_data, mode, duplicates=False):
                 if "South American" in [n1_ethnic, n2_ethnic]:
                     p_ethnic = "South American"
             elif player[3] in ["3", "6", "7", "8", "9"]:
-                p_ethnic = "African"
+                # SAMed with 7 is light-skinned
+                if "SAMed" in [n1_ethnic, n2_ethnic] and player[3] == "7":
+                    p_ethnic = "SAMed"
+                # South American with 7 is light-skinned
+                elif "South American" in [n1_ethnic, n2_ethnic] and player[3] == "7":
+                    p_ethnic = "South American"
+                else:
+                    p_ethnic = "African"
             elif player[3] == "10":
-                p_ethnic = "Asian"
+                # 1% of the South American population is Asian
+                if "South American" in [n1_ethnic, n2_ethnic] and random.randint(1, 100) != 1:
```

Comment:
I'd maybe get rid of the 1 percent randomness. Because this should be reflected by the distribution within FM itself. Plus it would also be easier to write tests for it.

Reply:
Sounds good. I'll make those changes and push them up.